sys/include/uuid.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_uuid
+ * @ingroup     sys
+ * @brief       Generate system wide unique IDs
```

Comment:
Maybe not applicable due to the more generalized nature of your API, but what about `devid` (or `dev_id`, so nobody will call it like some Germans call Davids ;-))